aip/general/0133.md

```diff
@@ -194,17 +175,32 @@ NOT_FOUND errors][permission-denied].
 - For ensuring idempotency in `Create` methods, see [AIP-155][].
 - For naming resources involving Unicode, see [AIP-210][].
 
+## Rationale
+
+### Requiring user-settable ids
+
+[IaC][] clients use the resource ID as a way to identify a resource for applying
```

Comment:
And other clients.

This basically replaces the approach recommended by https://google.aip.dev/155 to achieve idempotence for Create.

Using the resource id is simpler for clients and for services. It eliminates the need for a request id separate from the resource id. Resource ids already need to be unique within a project and location.

If the resource id isn't known until, say, a LRO completes, that widens the window during which a failure could cause the client to not know whether it had already issued the Create request and whether it had been successful or not. AIP 155 doesn't specify that the LRO needs to return the request id, either.

If the service needs resource ids to be of a particular form, it can enforce that with validation. We need better ways to convey such value constraints to users and clients, though.

Note that the desire for idempotence makes the practice of allowing the id to be optional fairly dubious. It would be pointless to make the resource id optional and then require an additional request id instead, for example.

Reply:
Those are great points! I'll add them to the PR. 

I think I buy the idea that we can do away with 155 entirely, and for those that want idempotency they need to provide a resource ID.

I think there's still good use cases for having an optional resource ID. e.g. cloud build triggers generating an ID on your behalf. But agreed that there should not be a concept of ensuring idempotency of an anonymous create request.

Reply:
That said I think removal of 155 needs be in a separate PR due to the size of the scope there: I've added #1103 for tracking.

Reply:
WDYT about me adding this to with the PR for AIP-155? I feel like we need to
make to change to remove 155 and replace it in lockstep, or else there's
conflicting guidance.